src/ray/core_worker/reference_count.cc

```diff
@@ -1042,6 +1042,34 @@ absl::optional<LocalityData> ReferenceCounter::GetLocalityData(
   return locality_data;
 }
 
+bool ReferenceCounter::ReportLocalityData(const ObjectID &object_id,
+                                          const absl::flat_hash_set<NodeID> &locations,
+                                          uint64_t object_size) {
+  absl::MutexLock lock(&mutex_);
+  auto it = object_id_refs_.find(object_id);
+  if (it == object_id_refs_.end()) {
+    RAY_LOG(INFO) << "Tried to report locality data for an object " << object_id
+                  << " that doesn't exist in the reference table."
+                  << " The object has probably already been freed.";
+    return false;
+  }
+  bool should_push = false;
+  if (!locations.empty()) {
+    should_push = true;
+    for (const auto &location : locations) {
+      it->second.locations.emplace(location);
+    }
+  }
+  if (object_size > 0) {
+    should_push = true;
+    it->second.object_size = object_size;
+  }
+  if (should_push) {
+    PushToLocationSubscribers(it);
```

Comment:
Do we need this? I thought this would only be for borrowers?

Probably good to have a check here that we are a borrower.

Reply:
🤦  That's a great point! This should always be a no-op, but nevertheless, we shouldn't be pushing here. And I'll be sure to add that check!